### PR TITLE
Add "No message timeout" braille setting

### PR DESCRIPTION
**Pull request by dkager**
In #6669 we discussed two corner cases for the braille message timeout:
1. To completely disable messages.
2. To show messages indefinitely.

In the end I decided to leave the original behavior alone: a message timeout of 0 disables messages altogether.
There is now an extra checkbox "No message timeout" to show messages indefinitely. It seemed most logical to add this checkbox after the edit field. Of course, enabling the checkbox disables the edit field.
This approach avoids having to add a config profile upgrade step and won't be confusing to users who currently have the timeout set to 0.

### CODE REVIEW

File: source/gui/settingsDialogs.py [hunk 1]
```diff
@@ -1562,14 +1562,22 @@ def makeSettings(self, settingsSizer):
 			self.shapeList.Disable()
 
 		# Translators: The label for a setting in braille settings to change how long a message stays on the braille display (in seconds).
-		messageTimeoutText = _("Message timeout (sec)")
+		messageTimeoutText = _("Message &timeout (sec)")
 		self.messageTimeoutEdit = sHelper.addLabeledControl(messageTimeoutText, nvdaControls.SelectOnFocusSpinCtrl,
 			min=int(config.conf.getConfigValidationParameter(["braille", "messageTimeout"], "min")),
 			max=int(config.conf.getConfigValidationParameter(["braille", "messageTimeout"], "max")),
 			initial=config.conf["braille"]["messageTimeout"])
+		if self.noMessageTimeoutCheckBox.GetValue():
```
Comment (feerrenrut) [inline source/gui/settingsDialogs.py]: This causes an exception, since `self.noMessageTimeoutCheckBox` is not created until line 1575

**Review by feerrenrut — CHANGES_REQUESTED** [review]
Thanks, overall this looks really good. What happens when the there is a message currently being displayed "indefinitely" and the setting is changed to a timeout?

**Comment by dkager** [comment]
@feerrenrut Pushed some new stuff.

**Comment by feerrenrut** [comment]
I have had a look at this, the alignment of the controls looks good. Could you please talk about what kind of testing you have done on this?

**Comment by dkager** [comment]
> Could you please talk about what kind of testing you have done on this?

Tested this with one display (since the code isn't driver-specific).
This was in early February, but from what I recall I used something like NVDA+t to have a quick message appear, then:
- Left it alone and verified that it stayed on the display.
- Focused another GUI element to make sure the message was cleared immediately.
- Used braille display keys to read the message and ensured it stayed on the display.
- Unselected the checkbox to make sure the message cleared (it does, because saving that setting causes the GUI to update which clears the message).

Also tested this with an actual use case, reading messages in Skype without having to rush.

**Review by feerrenrut — APPROVED** [review]

**Pull request merged**